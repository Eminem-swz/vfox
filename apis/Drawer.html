<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Drawer 抽屉类型弹窗 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.ebc52af7.css">
    <link rel="modulepreload" href="/vfox/assets/Home.84024184.js">
    <link rel="modulepreload" href="/vfox/assets/app.2bd60a78.js">
    <link rel="modulepreload" href="/vfox/assets/apis_Drawer.md.5d3ee718.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.2bd60a78.js">
    <meta name="twitter:title" content="Drawer 抽屉类型弹窗 | Vfox for Vue3">
    <meta property="og:title" content="Drawer 抽屉类型弹窗 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-7161a24b><div class="sidebar-button" data-v-7161a24b><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-7161a24b data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-7161a24b></div><div class="nav" data-v-7161a24b><nav class="nav-links" data-v-7161a24b data-v-15acbf05><!--[--><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/" data-v-641633f9>指南 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/design/" data-v-641633f9>设计 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/components/Button" data-v-641633f9>组件 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item active" href="/vfox/apis/LocalStorage" data-v-641633f9>API <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-dropdown-link" data-v-15acbf05 data-v-90f2c308><button class="button" data-v-90f2c308><span class="button-text" data-v-90f2c308>选择版本</span><span class="right button-arrow" data-v-90f2c308></span></button><ul class="dialog" data-v-90f2c308><!--[--><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>1.x</span><span class="icon" data-v-57d767ef><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-57d767ef><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item" href="/vfox/" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>2.x</span><span class="icon" data-v-57d767ef><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-6b49cdcd><nav class="nav-links nav" data-v-6b49cdcd data-v-15acbf05><!--[--><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/" data-v-641633f9>指南 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/design/" data-v-641633f9>设计 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/components/Button" data-v-641633f9>组件 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item active" href="/vfox/apis/LocalStorage" data-v-641633f9>API <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-dropdown-link" data-v-15acbf05 data-v-90f2c308><button class="button" data-v-90f2c308><span class="button-text" data-v-90f2c308>选择版本</span><span class="right button-arrow" data-v-90f2c308></span></button><ul class="dialog" data-v-90f2c308><!--[--><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>1.x</span><span class="icon" data-v-57d767ef><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-57d767ef><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item" href="/vfox/" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>2.x</span><span class="icon" data-v-57d767ef><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-6b49cdcd><!--[--><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/LocalStorage">LocalStorage</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Scroll">Scroll</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/Drawer">Drawer</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#picker-showpicker-object">Picker.showPicker(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#object-success-回调参数">object.success 回调参数</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#cascader-showcascader-object">Cascader.showCascader(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params-1">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-1">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#actionsheet-showactionsheet-object">ActionSheet.showActionSheet(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params-2">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-2">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#datepicker-showdatepicker-object">DatePicker.showDatePicker(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params-3">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#object-success-回调参数-3">object.success 回调参数</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-3">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#calendar-showcalendar-object">Calendar.showCalendar(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params-4">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#object-success-回调参数-4">object.success 回调参数</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-4">Usage</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><!--]--></ul><!--[--><!--]--></aside><!-- TODO: make this button accessible --><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="display:none;width:375px;height:667px;margin-top:1.5rem;" src="https://cdn.fox2.cn/2.x/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="drawer-抽屉类型弹窗"><a class="header-anchor" href="#drawer-抽屉类型弹窗" aria-hidden="true">#</a> Drawer 抽屉类型弹窗</h1><h2 id="picker-showpicker-object"><a class="header-anchor" href="#picker-showpicker-object" aria-hidden="true">#</a> Picker.showPicker(object)</h2><p>显示选择弹窗。</p><h3 id="params"><a class="header-anchor" href="#params" aria-hidden="true">#</a> Params</h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>弹窗标题</td></tr><tr><td>options</td><td>array</td><td>[]</td><td>否</td><td>数据集，参考 <a href="./../components/Picker.html#options-的结构">Picker options</a></td></tr><tr><td>value</td><td>string/number/string[]/number[]</td><td>[]</td><td>否</td><td>默认的选中项，参考 <a href="./../components/Picker.html#modelvalue-的结构">Picker modelValue</a></td></tr><tr><td>fieldNames</td><td>object</td><td>{ label: &#39;label&#39;, value: &#39;value&#39;, children: &#39;children&#39; }</td><td>否</td><td>自定义 options 中 label value children 的字段 key</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h3 id="object-success-回调参数"><a class="header-anchor" href="#object-success-回调参数" aria-hidden="true">#</a> object.success 回调参数</h3><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定，此时返回 <code>detail</code></td></tr><tr><td>detail?.value</td><td>number[]/string[]</td><td>[&quot;zaolei&quot;, &quot;lunzao&quot;]</td></tr><tr><td>detail?.valueString</td><td>string</td><td>&quot;zaolei/lunzao&quot;</td></tr><tr><td>detail?.label</td><td>string[]</td><td>[&quot;藻类&quot;, &quot;轮藻&quot;]</td></tr><tr><td>detail?.labelString</td><td>string</td><td>&quot;藻类/轮藻&quot;</td></tr></tbody></table><h3 id="usage"><a class="header-anchor" href="#usage" aria-hidden="true">#</a> Usage</h3><div class="language-"><pre><code>import { Picker } from &#39;vfox&#39;

Picker.showPicker({
  title: &#39;Picker&#39;,
  options: multiOptions,
  success: ({ confirm, cancel, detail }) =&gt; {
    ...
  }
})
</code></pre></div><h2 id="cascader-showcascader-object"><a class="header-anchor" href="#cascader-showcascader-object" aria-hidden="true">#</a> Cascader.showCascader(object)</h2><p>显示级联选择弹窗。</p><h3 id="params-1"><a class="header-anchor" href="#params-1" aria-hidden="true">#</a> Params</h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>没有选择时弹窗的标题文案</td></tr><tr><td>options</td><td>array</td><td>[]</td><td>否</td><td>数据集，参考 <a href="./../components/Cascader.html#options-的结构">Cascader options</a></td></tr><tr><td>value</td><td>string/number/string[]/number[]</td><td>[]</td><td>否</td><td>默认的选中项，考参考 <a href="./../components/Cascader.html#modelvalue-的结构">Cascader modelValue</a></td></tr><tr><td>fieldNames</td><td>object</td><td>{ label: &#39;label&#39;, value: &#39;value&#39;, children: &#39;children&#39; }</td><td>否</td><td>自定义 options 中 label value children 的字段 key</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h4 id="object-success-回调参数-1"><a class="header-anchor" href="#object-success-回调参数-1" aria-hidden="true">#</a> object.success 回调参数</h4><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定，此时返回 <code>detail</code></td></tr><tr><td>detail?.value</td><td>number[]/string[]</td><td>[&quot;zaolei&quot;, &quot;lunzao&quot;]</td></tr><tr><td>detail?.valueString</td><td>string</td><td>&quot;zaolei/lunzao&quot;</td></tr><tr><td>detail?.label</td><td>string[]</td><td>[&quot;藻类&quot;, &quot;轮藻&quot;]</td></tr><tr><td>detail?.labelString</td><td>string</td><td>&quot;藻类/轮藻&quot;</td></tr></tbody></table><h3 id="usage-1"><a class="header-anchor" href="#usage-1" aria-hidden="true">#</a> Usage</h3><div class="language-"><pre><code>import { Cascader } from &#39;vfox&#39;

Cascader.showCascader({
  title: &#39;植物&#39;,
  options: cascadeOptions,
  success: ({ confirm, cancel, detail }) =&gt; {
    ...
  }
})
</code></pre></div><h2 id="actionsheet-showactionsheet-object"><a class="header-anchor" href="#actionsheet-showactionsheet-object" aria-hidden="true">#</a> ActionSheet.showActionSheet(object)</h2><p>显示动作面板。</p><h3 id="params-2"><a class="header-anchor" href="#params-2" aria-hidden="true">#</a> Params</h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>提示的标题，不设置则不展示</td></tr><tr><td>options</td><td>array</td><td></td><td>否</td><td>选项列表，参考 <a href="./../components/ActionSheet.html#options-的结构">ActionSheet options</a></td></tr><tr><td>showCancel</td><td>boolean</td><td>false</td><td>否</td><td>是否显示取消按钮</td></tr><tr><td>cancelText</td><td>string</td><td>&#39;取消&#39;</td><td>否</td><td>取消按钮的文字</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，选择选项）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h4 id="object-success-回调参数-2"><a class="header-anchor" href="#object-success-回调参数-2" aria-hidden="true">#</a> object.success 回调参数</h4><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示用户点击了选项，此时返回 <code>detail</code></td></tr><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示用户点击了取消</td></tr><tr><td>detail?.item</td><td>any</td><td>options[index] 的副本</td></tr><tr><td>detail?.index</td><td>number</td><td>索引</td></tr></tbody></table><h3 id="usage-2"><a class="header-anchor" href="#usage-2" aria-hidden="true">#</a> Usage</h3><div class="language-"><pre><code>import { ActionSheet } from &#39;vfox&#39;

ActionSheet.showActionSheet({
  title: &#39;标题&#39;,
  options: [
    {
      name: &#39;选项1&#39;,
      description: &#39;选项1的描述文案&#39;
    },
    {
      name: &#39;选项2&#39;
    },
    {
      name: &#39;选项3&#39;
    }
  ],
  showCancel: true,
  success: ({ confirm, cancel, detail }) =&gt; {
    ...
  }
})
</code></pre></div><h2 id="datepicker-showdatepicker-object"><a class="header-anchor" href="#datepicker-showdatepicker-object" aria-hidden="true">#</a> DatePicker.showDatePicker(object)</h2><p>日期选择弹窗。</p><h3 id="params-3"><a class="header-anchor" href="#params-3" aria-hidden="true">#</a> Params</h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>弹窗标题</td></tr><tr><td>value</td><td>Date</td><td></td><td>否</td><td>默认选择值</td></tr><tr><td>mode</td><td>string</td><td></td><td>否</td><td>选择的模式，可选 &#39;date&#39;, &#39;time&#39;, &#39;datetime&#39; 等，参考 <a href="./../components/DatePicker.html#initialmode-的合法值">DatePicker initialMode</a></td></tr><tr><td>minDate</td><td>Date</td><td>当天对应 10 年前的 Date 实例</td><td>否</td><td>最小时间对应的 Date 实例（含）</td></tr><tr><td>maxDate</td><td>Date</td><td>当天最后一秒的 Date 实例</td><td>否</td><td>最大时间对应的 Date 实例（含）</td></tr><tr><td>filter</td><td>Function</td><td></td><td>否</td><td>选项过滤器，<code>filter(number: number, type: &#39;year&#39; | &#39;month&#39; | &#39;day&#39; | &#39;hour&#39; | &#39;minute&#39; | &#39;second&#39;) =&gt; boolean</code></td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h3 id="object-success-回调参数-3"><a class="header-anchor" href="#object-success-回调参数-3" aria-hidden="true">#</a> object.success 回调参数</h3><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定，此时返回 <code>detail</code></td></tr><tr><td>detail?.formatted</td><td>string</td><td>&quot;2021-05-01&quot;</td></tr><tr><td>detail?.value</td><td>Date</td><td>对应日期的 Date 实例</td></tr><tr><td>detail?.valueArray</td><td>number[]</td><td>[2021, 5, 1]</td></tr></tbody></table><h3 id="usage-3"><a class="header-anchor" href="#usage-3" aria-hidden="true">#</a> Usage</h3><div class="language-"><pre><code>import { DatePicker } from &#39;vfox&#39;

DatePicker.showDatePicker({
  title: &#39;DatePicker&#39;,
  success: ({ confirm, cancel, detail }) =&gt; {
    ...
  }
})
</code></pre></div><h2 id="calendar-showcalendar-object"><a class="header-anchor" href="#calendar-showcalendar-object" aria-hidden="true">#</a> Calendar.showCalendar(object)</h2><p>日历选择弹窗。</p><h3 id="params-4"><a class="header-anchor" href="#params-4" aria-hidden="true">#</a> Params</h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>mode</td><td>string</td><td>&#39;single&#39;</td><td>否</td><td>模式：<code>single</code> 表示选择一天，<code>range</code> 选择一个日期区间</td></tr><tr><td>title</td><td>string</td><td></td><td>否</td><td>弹窗标题</td></tr><tr><td>value</td><td>Date/Date[]</td><td>[]</td><td>否</td><td>默认选择值，<code>range</code> 模式下需要提供两个</td></tr><tr><td>minDate</td><td>Date</td><td>当前日期</td><td>否</td><td>可选最小值</td></tr><tr><td>maxDate</td><td>Date</td><td>当前日期的六个月后</td><td>否</td><td>可选最大值</td></tr><tr><td>allowSameDay</td><td>boolean</td><td>false</td><td>否</td><td><code>range</code> 模式生效，设置开始结束时间是否可以同一天</td></tr><tr><td>maxRange</td><td>number</td><td>Infinity</td><td>否</td><td><code>range</code> 模式生效，选择区间的最长天数</td></tr><tr><td>dayHandler</td><td>Function</td><td></td><td>否</td><td>日历每个日期处理函数，参考 <a href="./../components/Calendar.html#dayhandler-函数">Calendar dayHandler</a></td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h3 id="object-success-回调参数-4"><a class="header-anchor" href="#object-success-回调参数-4" aria-hidden="true">#</a> object.success 回调参数</h3><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定，此时返回 <code>detail</code></td></tr><tr><td>detail?.value</td><td>Date[]</td><td>开始 1 个 Date 实例 或 开始和结束 2 个 Date 实例</td></tr><tr><td>detail?.valueArray</td><td>number[][]</td><td>如：[[2021, 5, 1], [2021, 5, 30]]</td></tr><tr><td>detail?.formatted</td><td>string</td><td>05-21 ~ 05-22</td></tr><tr><td>detail?.rangeCount</td><td>number</td><td>选择区间持续的天数（含首尾）</td></tr></tbody></table><h3 id="usage-4"><a class="header-anchor" href="#usage-4" aria-hidden="true">#</a> Usage</h3><div class="language-"><pre><code>import { Calendar } from &#39;vfox&#39;

Calendar.showCalendar({
  type: &#39;range&#39;,
  showClose: true,
  success: ({ confirm, cancel, detail }) =&gt; {
    ...
  }
})
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/apis/Modal" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Modal</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/apis/Popover" data-v-38ede35f><span class="text" data-v-38ede35f>Popover</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"index.md\":\"b4e941d0\",\"apis_drawer.md\":\"5d3ee718\",\"apis_intersectionobserver.md\":\"3857133e\",\"apis_localstorage.md\":\"1257b295\",\"apis_modal.md\":\"8d894961\",\"apis_notify.md\":\"f2df1a16\",\"apis_popover.md\":\"0f623be3\",\"apis_preview.md\":\"91583ee2\",\"apis_scroll.md\":\"c27c9f01\",\"apis_selectorquery.md\":\"7a0b9932\",\"apis_toast.md\":\"27487c67\",\"apis_index.md\":\"e3a0097d\",\"components_actionsheet.md\":\"00ca51f2\",\"components_activityindicator.md\":\"9b6d3366\",\"components_avatar.md\":\"01ab8ae2\",\"components_backtop.md\":\"77055cdb\",\"components_badge.md\":\"8a555d8b\",\"components_button.md\":\"1910d3f9\",\"components_calendar.md\":\"00a8fb52\",\"components_cascader.md\":\"695c5614\",\"components_cell.md\":\"79b0aac7\",\"components_checkbox.md\":\"5e51b6a2\",\"components_circleprogress.md\":\"3f5f5851\",\"components_collapse.md\":\"3b9f1141\",\"components_copy.md\":\"7ba467db\",\"components_countdown.md\":\"b2680b9a\",\"components_countup.md\":\"2e75d992\",\"components_datepicker.md\":\"f58c87d4\",\"components_dialog.md\":\"9991c740\",\"components_divider.md\":\"cb67333c\",\"components_drawer.md\":\"02e0cb4e\",\"components_dropdown.md\":\"a341e9da\",\"components_empty.md\":\"586cb228\",\"components_fixed.md\":\"a1018e8d\",\"components_flatlist.md\":\"838713de\",\"components_form.md\":\"f25b6809\",\"components_group.md\":\"b966984a\",\"components_icon.md\":\"01e5b298\",\"components_image.md\":\"14e16ca6\",\"components_imagepreview.md\":\"4aac496e\",\"components_imageuploader.md\":\"d438901a\",\"components_indexview.md\":\"6597421b\",\"components_input.md\":\"1e991179\",\"components_layout.md\":\"717db433\",\"components_loadmore.md\":\"2b97fde5\",\"components_modal.md\":\"3baeadb9\",\"components_navbar.md\":\"1ecdcfbf\",\"components_noticebar.md\":\"2d41effe\",\"components_notify.md\":\"6209aa98\",\"components_numberkeyboard.md\":\"b887a913\",\"components_order.md\":\"f8549cfa\",\"components_pagination.md\":\"a5249f17\",\"components_picker.md\":\"e82123bf\",\"components_popdialog.md\":\"9d845cfb\",\"components_popmenu.md\":\"f3243d5b\",\"components_popover.md\":\"634c0ee0\",\"components_price.md\":\"e5b78306\",\"components_progress.md\":\"c3150621\",\"components_qrcode.md\":\"cdf3f859\",\"components_radio.md\":\"22cdc0fd\",\"components_range.md\":\"37fb3230\",\"components_rate.md\":\"edd513e5\",\"components_result.md\":\"a0a4c081\",\"components_scrolltab.md\":\"33000c23\",\"components_scrollview.md\":\"3b60298e\",\"components_searchbar.md\":\"e6c8bca7\",\"components_sidetab.md\":\"e77fff35\",\"components_skeleton.md\":\"c01adb11\",\"components_slider.md\":\"f8127b11\",\"components_stepper.md\":\"903581e9\",\"components_steps.md\":\"1cc6fe4a\",\"components_sticky.md\":\"0c230008\",\"components_stopwatch.md\":\"01182634\",\"components_swipecell.md\":\"0e3d18e9\",\"components_swiper.md\":\"8a20556d\",\"components_switch.md\":\"7d6866c7\",\"components_tab.md\":\"cee117ea\",\"components_tabbar.md\":\"f3cf4f1f\",\"components_tabview.md\":\"e9793808\",\"components_tag.md\":\"50adf7ce\",\"components_timeago.md\":\"4b27a4af\",\"components_timeline.md\":\"d22b6010\",\"components_toast.md\":\"ec52a06e\",\"components_index.md\":\"de9625df\",\"design_index.md\":\"2680ae1e\"}")</script>
    <script type="module" async src="/vfox/assets/app.2bd60a78.js"></script>
  </body>
</html>