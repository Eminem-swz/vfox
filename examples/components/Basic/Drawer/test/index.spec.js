/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Drawer from '@/components/Drawer'

describe('Drawer', () => {
  test('should render prop title="Drawer" correctly', () => {
    const wrapper = mount(Drawer, {
      propsData: {
        title: 'Drawer'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
