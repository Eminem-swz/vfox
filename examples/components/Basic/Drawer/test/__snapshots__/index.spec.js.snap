// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render prop title="Drawer" correctly 1`] = `
<!--teleport start-->
<!--teleport end-->
`;
